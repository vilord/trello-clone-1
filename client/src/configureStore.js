import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import reducer from './reducers';

const middleware = [thunk];
if (process.env.NODE_ENV !== 'production') {
  middleware.push(createLogger());
}

export default preloadedState => {
  return createStore(
    reducer,
    preloadedState,
    composeWithDevTools(applyMiddleware(...middleware)),
  );
};
