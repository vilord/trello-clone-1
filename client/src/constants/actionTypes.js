/**
 * USER
 */
export const SIGNUP_USER_REQUEST = 'SIGNUP_USER_REQUEST';
export const SIGNUP_USER_SUCCESS = 'SIGNUP_USER_SUCCESS';
export const SIGNUP_USER_FAILURE = 'SIGNUP_USER_FAILURE';

export const LOGIN_USER_REQUEST = 'LOGIN_USER_REQUEST';
export const LOGIN_USER_SUCCESS = 'LOGIN_USER_SUCCESS';
export const LOGIN_USER_FAILURE = 'LOGIN_USER_FAILURE';

export const LOGOUT_USER = 'LOGOUT_USER';

export const GET_USER_SESSION_REQUEST = 'GET_USER_SESSION_REQUEST';
export const GET_USER_SESSION_SUCCESS = 'GET_USER_SESSION_SUCCESS';

export const SET_USER_PROFILE_REQUEST = 'SET_USER_PROFILE_REQUEST';
export const SET_USER_PROFILE_SUCCESS = 'SET_USER_PROFILE_SUCCESS';
export const SET_USER_PROFILE_FAILURE = 'SET_USER_PROFILE_FAILURE';

/**
 * UI
 */
export const SET_UI_ERROR = 'SET_UI_ERROR';
export const RESET_UI_ERROR = 'RESET_UI_ERROR';

export const SHOW_BOARDS_EXPLORER = 'SHOW_BOARDS_EXPLORER';
export const HIDE_BOARDS_EXPLORER = 'HIDE_BOARDS_EXPLORER';

export const FOCUS_HEADER_SEARCH = 'FOCUS_HEADER_SEARCH';
export const BLUR_HEADER_SEARCH = 'BLUR_HEADER_SEARCH';

export const SHOW_CREATE_MENU = 'SHOW_CREATE_MENU';
export const HIDE_CREATE_MENU = 'HIDE_CREATE_MENU';

export const SHOW_NOTIFICATIONS = 'SHOW_NOTIFICATIONS';
export const HIDE_NOTIFICATIONS = 'HIDE_NOTIFICATIONS';

export const SHOW_USER_MENU = 'SHOW_USER_MENU';
export const HIDE_USER_MENU = 'HIDE_USER_MENU';

export const SHOW_BOARD_MENU = 'SHOW_BOARD_MENU';
export const HIDE_BOARD_MENU = 'HIDE_BOARD_MENU';

/**
 * Cards
 */
export const SHOW_CARD_DETAIL = 'SHOW_CARD_DETAIL ';
export const HIDE_CARD_DETAIL = 'HIDE_CARD_DETAIL ';

export const SHOW_LABEL_TEXT = 'SHOW_LABEL_TEXT';
export const HIDE_LABEL_TEXT = 'HIDE_LABEL_TEXT';
