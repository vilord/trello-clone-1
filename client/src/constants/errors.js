export const invalidEmail = {
  kind: 'invalidEmail',
  header: 'Verify email!',
  message: "That doesn't look like an email addressâ€¦",
};

export const shortPassword = {
  kind: 'shortPassword',
  header: 'Password is too short!',
  message: 'Your password must be at least 8 characters long.',
};

export const userExists = {
  kind: 'userExists',
  header: 'User already exists!',
  message: 'Email is already in use by another account.'
};

export const wrongCredentials ={
  kind: 'wrongCredentials',
  header: 'Something wrong!',
  message: 'Incorrect email (username) or password.'
};

export const serverError = {
  kind: 'serverError',
  header: 'Unknown Server Error!',
  message: 'Our server may be down. Please retry later!'
};
