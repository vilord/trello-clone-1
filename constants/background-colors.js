module.exports = {
  BLUE: '#127ABD',
  ORANGE: '#D18F3E',
  GREEN: '#53973F',
  RED: '#AE4736',
  PURPLE: '#88629C',
  PINK: '#CB5C91',
  LIGHT_GREEN: '#50BE6E',
  LIGHT_BLUE: '#1CAECA',
  GREY: '#838C91',
};
