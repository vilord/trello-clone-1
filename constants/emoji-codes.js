module.exports = [
  'thumbsup',
  'thumbsdown',
  'raised_hands',
  'pray',
  'eyes',
  'rocket',
  'joy',
  'blush',
  'hearth_eyes',
  'throphy',
  'heart',
  'broken_heart',
];
