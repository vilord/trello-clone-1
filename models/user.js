const mongoose = require('mongoose');
const Schema = mongoose.Schema;
const { ObjectId } = Schema.Types;
const isEmail = require('validator/lib/isEmail');
const isUrl = require('validator/lib/isUrl');

const UserSchema = new Schema(
  {
    username: {
      type: String,
      unique: true,
      trim: true,
      match: /^((?!\s+).)*$/,
      required: true,
    },
    password: {
      type: String,
      minlength: 6,
      maxlength: 256,
      // TODO: password match
      // match: /^.*[a-zA-Z].*[0-9]+.*$/,
      bcrypt: true,
      required: function() {
        return !this.google_id;
      },
    },
    google_id: {
      type: Number,
      required: function() {
        return !this.password;
      },
    },
    email: {
      type: String,
      unique: true,
      trim: true,
      lowercase: true,
      required: true,
      validate: {
        validator: x => isEmail(x),
        message: 'Must be a valid email address',
      },
    },
    name: {
      type: String,
      required: true,
      trim: true,
      match: /^((?!\s{2,}).)*$/,
      maxlength: 70,
    },
    initials: {
      type: String,
      uppercase: true,
      maxlength: 4,
      minlength: 1,
      match: /^((?!\s).)*$/,
    },
    bio: {
      type: String,
      trim: true,
      minlength: 1,
      maxlength: 500,
    },
    avatar: {
      type: String,
      trim: true,
      validate: {
        validator: x => isUrl(x),
        message: 'it needs a valid URL',
      },
    },
    assigned_cards: [
      {
        type: ObjectId,
        ref: 'Card',
      },
    ],
    boards: [
      {
        board: {
          type: ObjectId,
          ref: 'Board',
        },
        favorite: {
          type: Boolean,
          default: false,
        },
      },
    ],
    teams: [
      {
        type: ObjectId,
        ref: 'Team',
      },
    ],
    activity: [
      {
        type: ObjectId,
        ref: 'Activity',
      },
    ],
  },
  {
    timestamps: true,
  },
);

/**
 * Password Hashing and Salting Plugin
 */
UserSchema.plugin(require('mongoose-bcrypt'), { rounds: 8 });

/**
 * Remove sensitive information by default
 * when transforming the model toObject.
 */
UserSchema.options.toObject = {
  transform: function(doc, ret, options) {
    delete ret._id;
    delete ret.__v;
    delete ret.password;
    delete ret.createdAt;
    delete ret.updatedAt;
    delete ret.google_id;
    return ret;
  },
};

/**
 * Pre validation hook
 */
UserSchema.pre('validate', async function(next) {
  if (!this.name) {
    const err = new Error('User validation failed: name is required.');
    err.name = 'ValidationError';
    return next(err);
  }

  // Generates unique username based on the name.
  const genUsername = async name => {
    let username = name.replace(/\s/g, '').toLowerCase();
    const count = await this.model('User')
      .find({ username: { $regex: username } })
      .lean()
      .count();
    username += count === 0 ? '' : count;
    return username;
  };

  this.username = this.username || (await genUsername(this.name));

  next();
});

module.exports = mongoose.model('User', UserSchema);
