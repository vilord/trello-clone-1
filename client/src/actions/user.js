import * as types from '../constants/actionTypes';

/*
 * Signup User
 */
export const signupUserRequest = () => ({
  type: types.SIGNUP_USER_REQUEST,
});

export const signupUserFailure = error => ({
  type: types.SIGNUP_USER_FAILURE,
  error,
});

/**
 * Login User
 */
export const loginUserRequest = () => ({
  type: types.LOGIN_USER_REQUEST,
});

export const loginUserSuccess = user => ({
  type: types.LOGIN_USER_SUCCESS,
  user,
});

export const loginUserFailure = error => ({
  type: types.LOGIN_USER_FAILURE,
  error,
});

/**
 * Logout User
 */
export const logoutUser = () => ({
  type: types.LOGOUT_USER,
});

/**
 * Set User Profile
 */
export const setUserProfileRequest = () => ({
  type: types.SET_USER_PROFILE_REQUEST,
});

export const setUserProfileSuccess = profile => ({
  type: types.SET_USER_PROFILE_SUCCESS,
  profile,
});

export const setUserProfileFailure = error => ({
  type: types.SET_USER_PROFILE_FAILURE,
  error,
});

/**
 * Async Actions
 */
export const signupUser = (newUser, history) => async dispatch => {
  dispatch(signupUserRequest());

  try {
    const res = await fetch(`/users`, {
      method: 'POST',
      credentials: 'include',
      headers: {
        'content-type': 'application/json',
      },
      body: JSON.stringify(newUser),
    });

    const { user, error } = await res.json();

    if (res.status === 200 && user) {
      return dispatch(loginUserSuccess(user));
    }

    return dispatch(signupUserFailure(error));
  } catch (err) {
    console.log(err);
  }
};

export const loginUser = (userLoginInfo, history) => async dispatch => {
  dispatch(loginUserRequest());

  try {
    const res = await fetch(`/auth/login`, {
      method: 'POST',
      headers: {
        'content-type': 'application/json',
      },
      body: JSON.stringify(userLoginInfo),
      credentials: 'include',
    });

    const { user, error } = await res.json();

    if (res.status === 200 && user) {
      return dispatch(loginUserSuccess(user));
    }

    return dispatch(loginUserFailure(error));
  } catch (err) {
    console.log(err);
  }
};

export const getUserSession = () => async dispatch => {
  try {
    const res = await fetch('/auth/user-session', {
      credentials: 'include',
      headers: {
        'content-type': 'application/json',
      },
    });

    const { user } = await res.json();

    if (res.status === 200 && user) {
      return dispatch(loginUserSuccess(user));
    }

    return dispatch(logoutUser());
  } catch (err) {
    console.log(err);
  }
};

export const sendLogoutUser = () => async dispatch => {
  try {
    const res = await fetch('/auth/logout', {
      method: 'POST',
      credentials: 'inlude',
    });

    if (res.status !== 200) throw new Error(await res.json());
  } catch (err) {
    console.log(err);
  }

  dispatch(logoutUser());
};

export const setUserProfile = newProfile => async dispatch => {
  dispatch(setUserProfileRequest());

  try {
    const res = await fetch('/users/profile', {
      method: 'PUT',
      credentials: 'include',
      body: {
        profile: newProfile,
      },
    });

    const { profile, error } = await res.json();

    if (res.status === 200 && profile) {
      return dispatch(setUserProfileSuccess(profile));
    }

    return dispatch(setUserProfileFailure(error));
  } catch (err) {
    console.log(err);
  }
};
