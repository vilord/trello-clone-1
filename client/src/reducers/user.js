import * as types from '../constants/actionTypes';
import initUserState from './initUserState';

const userReducer = (state = initUserState, action) => {
  switch (action.type) {
    case types.SIGNUP_USER_SUCCESS:
      return {
        ...state,
        ...action.user,
      };
    case types.GET_USER_SESSION_SUCCESS:
    case types.LOGIN_USER_SUCCESS:
      return {
        ...state,
        ...action.user,
      };
    case types.SET_USER_PROFILE_SUCCESS:
      return {
        ...state,
        ...action.profile,
      };
    default:
      return state;
  }
};

/**
 * Higher order function that resets
 * the state to its initial value on
 * LOGOUT_USER action.
 */
const logoutWrapper = (state, action) => {
  if (action.type === types.LOGOUT_USER) {
    state = undefined;
  }
  return userReducer(state, action);
};

export default logoutWrapper;
