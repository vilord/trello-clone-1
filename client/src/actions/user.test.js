import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import fetchMock from 'fetch-mock';

import * as actions from './user';
import * as types from '../constants/actionTypes';

const middleware = [thunk];
const mockStore = configureStore(middleware);

describe('user actions', () => {
  describe('Action Creators', () => {
    describe('Logout User', () => {
      it('creates a LOGOUT_USER action', () => {
        const expected = { type: types.LOGOUT_USER };
        expect(actions.logoutUser()).toEqual(expected);
      });
    });

    describe('Signup User', () => {
      it('creates a SIGNUP_USER_REQUEST', () => {
        const expected = { type: types.SIGNUP_USER_REQUEST };
        expect(actions.signupUserRequest()).toEqual(expected);
      });

      it('creates a SIGNUP_USER_FAILURE action', () => {
        const expected = { type: types.SIGNUP_USER_FAILURE };
        expect(actions.signupUserFailure()).toEqual(expected);
      });
    });

    describe('Login User', () => {
      it('creates a LOGIN_USER_REQUEST action', () => {
        const expected = { type: types.LOGIN_USER_REQUEST };
        expect(actions.loginUserRequest()).toEqual(expected);
      });

      it('creates a LOGIN_USER_SUCCESS action', () => {
        const user = { username: 'juanito' };
        const expected = { type: types.LOGIN_USER_SUCCESS, user };
        expect(actions.loginUserSuccess(user)).toEqual(expected);
      });

      it('creates a LOGIN_USER_FAILURE action', () => {
        const error = new Error('Login user failure');
        const expected = { type: types.LOGIN_USER_FAILURE, error };
        expect(actions.loginUserFailure(error)).toEqual(expected);
      });
    });

    describe('Set User Profile', () => {
      it('creates a SET_USER_PROFILE_REQUEST action', () => {
        const expected = { type: types.SET_USER_PROFILE_REQUEST };
        expect(actions.setUserProfileRequest()).toEqual(expected);
      });

      it('creates a SET_USER_PROFILE_SUCCESS action', () => {
        const expected = { type: types.SET_USER_PROFILE_SUCCESS };
        expect(actions.setUserProfileSuccess()).toEqual(expected);
      });

      it('creates a SET_USER_PROFILE_FAILURE action', () => {
        const expected = { type: types.SET_USER_PROFILE_FAILURE };
        expect(actions.setUserProfileFailure()).toEqual(expected);
      });
    });
  });

  describe('Async Actions', () => {
    afterEach(() => {
      fetchMock.reset();
      fetchMock.restore();
    });

    describe('signupUser', () => {
      const user = { username: 'johndoe' };

      it('on success', () => {
        fetchMock.post('/users', {
          headers: { 'content-type': 'application/json' },
          body: {
            user,
          },
        });

        const expected = [
          { type: types.SIGNUP_USER_REQUEST },
          { type: types.LOGIN_USER_SUCCESS, user },
        ];

        const store = mockStore({});

        return store.dispatch(actions.signupUser(user)).then(() => {
          expect(store.getActions()).toEqual(expected);
        });
      });

      it('on failure', () => {
        const res = {
          error: 'Signup failure.',
        };
        fetchMock.post('/users', {
          status: 409,
          body: res,
        });

        const expected = [
          { type: types.SIGNUP_USER_REQUEST },
          { type: types.SIGNUP_USER_FAILURE, error: res.error },
        ];

        const store = mockStore({});

        return store.dispatch(actions.signupUser(user)).then(() => {
          expect(store.getActions()).toEqual(expected);
        });
      });
    });

    describe('loginUser', () => {
      const user = { username: 'johndoe' };

      it('on success', () => {
        fetchMock.post('/auth/login', {
          headers: { 'content-type': 'application/json' },
          body: { user },
        });

        const expected = [
          { type: types.LOGIN_USER_REQUEST },
          { type: types.LOGIN_USER_SUCCESS, user },
        ];

        const store = mockStore({});

        return store.dispatch(actions.loginUser(user)).then(() => {
          expect(store.getActions()).toEqual(expected);
        });
      });

      it('on failure', () => {
        const res = {
          error: 'loginUser failure.',
        };

        fetchMock.post('/auth/login', {
          status: 400,
          body: res,
        });

        const expected = [
          { type: types.LOGIN_USER_REQUEST },
          { type: types.LOGIN_USER_FAILURE, error: res.error },
        ];

        const store = mockStore({});

        return store.dispatch(actions.loginUser(user)).then(() => {
          expect(store.getActions()).toEqual(expected);
        });
      });
    });

    describe('getUserSession', () => {
      it('on success', () => {
        const user = {
          name: 'John Doe',
          email: 'johndoe@example.com',
          password: '0notJohnDoe',
        };

        fetchMock.get('/auth/user-session', {
          headers: { 'content-type': 'application/json' },
          body: {
            user,
          },
        });

        const expected = [{ type: types.LOGIN_USER_SUCCESS, user }];

        const store = mockStore({ user });

        return store.dispatch(actions.getUserSession()).then(() => {
          expect(store.getActions()).toEqual(expected);
        });
      });

      it('on failure', () => {
        fetchMock.get('/auth/user-session', {
          status: 400,
          body: {
            error: 'getUserSession failure.',
          },
        });

        const expected = [{ type: types.LOGOUT_USER }];

        const store = mockStore({});

        return store.dispatch(actions.getUserSession()).then(() => {
          expect(store.getActions()).toEqual(expected);
        });
      });
    });

    describe('sendLogoutUser', () => {
      it('on success', () => {
        fetchMock.post('/auth/logout', {});

        const expected = [{ type: types.LOGOUT_USER }];

        const store = mockStore({});

        return store.dispatch(actions.sendLogoutUser()).then(() => {
          expect(store.getActions()).toEqual(expected);
        });
      });
    });

    describe('setUserProfile', () => {
      it('on success', () => {
        const profile = {
          name: 'John Doe',
          username: 'johndoe',
          initials: 'JD',
          bio: 'Web Developer',
          avatar: 'https://example.com/profile_pic',
        };

        fetchMock.put('/users/profile', {
          headers: { 'content-type': 'application/json' },
          body: {
            profile,
          },
        });

        const expected = [
          { type: types.SET_USER_PROFILE_REQUEST },
          { type: types.SET_USER_PROFILE_SUCCESS, profile },
        ];

        const store = mockStore({});

        return store.dispatch(actions.setUserProfile(profile)).then(() => {
          expect(store.getActions()).toEqual(expected);
        });
      });

      it('on failure', () => {
        const  error= 'setUserProfile failure.';

        fetchMock.put('/users/profile', {
          status: 400,
          body: { error }
        });

        const expected = [
          { type: types.SET_USER_PROFILE_REQUEST },
          { type: types.SET_USER_PROFILE_FAILURE, error },
        ];

        const store = mockStore({});

        const profile = { username: 'johndoe' };
        return store.dispatch(actions.setUserProfile(profile)).then(() => {
          expect(store.getActions()).toEqual(expected);
        });
      });
    });
  });
});
