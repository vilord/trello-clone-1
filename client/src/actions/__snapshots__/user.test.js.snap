// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Actions Action Creators Login User creates a LOGIN_USER_FAILURE action 1`] = `
Object {
  "error": Object {
    "header": "Something wrong!",
    "kind": "wrongCredentials",
    "message": "Incorrect email (username) or password.",
  },
  "type": "LOGIN_USER_FAILURE",
}
`;

exports[`User Actions Action Creators Login User creates a LOGIN_USER_REQUEST action 1`] = `
Object {
  "type": "LOGIN_USER_REQUEST",
}
`;

exports[`User Actions Action Creators Login User creates a LOGIN_USER_SUCCESS action 1`] = `
Object {
  "type": "LOGIN_USER_SUCCESS",
  "user": Object {
    "email": "johndoe@example.com",
    "username": "johndoe",
  },
}
`;

exports[`User Actions Action Creators Logout User creates a LOGOUT_USER action 1`] = `
Object {
  "type": "LOGOUT_USER",
}
`;

exports[`User Actions Action Creators Set User Profile creates a SET_USER_PROFILE_FAILURE action 1`] = `
Object {
  "error": Object {
    "header": "Unknown Server Error!",
    "kind": "serverError",
    "message": "Our server may be down. Please retry later!",
  },
  "type": "SET_USER_PROFILE_FAILURE",
}
`;

exports[`User Actions Action Creators Set User Profile creates a SET_USER_PROFILE_REQUEST action 1`] = `
Object {
  "type": "SET_USER_PROFILE_REQUEST",
}
`;

exports[`User Actions Action Creators Set User Profile creates a SET_USER_PROFILE_SUCCESS action 1`] = `
Object {
  "profile": Object {
    "initials": "JD",
    "name": "John Doe",
  },
  "type": "SET_USER_PROFILE_SUCCESS",
}
`;

exports[`User Actions Action Creators Signup User creates a SIGNUP_USER_FAILURE 1`] = `
Object {
  "error": Object {
    "header": "Unknown Server Error!",
    "kind": "serverError",
    "message": "Our server may be down. Please retry later!",
  },
  "type": "SIGNUP_USER_FAILURE",
}
`;

exports[`User Actions Action Creators Signup User creates a SIGNUP_USER_REQUEST 1`] = `
Object {
  "type": "SIGNUP_USER_REQUEST",
}
`;

exports[`User Actions Action Creators Signup User creates a SIGNUP_USER_SUCCESS 1`] = `
Object {
  "type": "SIGNUP_USER_SUCCESS",
  "user": Object {
    "email": "johndoe@example.com",
    "username": "johndoe",
  },
}
`;

exports[`User Actions Async Actions getUserSession on failure 1`] = `"/login"`;

exports[`User Actions Async Actions getUserSession on failure 2`] = `
Array [
  Object {
    "type": "GET_USER_SESSION_REQUEST",
  },
  Object {
    "type": "LOGOUT_USER",
  },
]
`;

exports[`User Actions Async Actions getUserSession on success 1`] = `
Array [
  Object {
    "type": "GET_USER_SESSION_REQUEST",
  },
  Object {
    "type": "GET_USER_SESSION_SUCCESS",
    "user": Object {
      "email": "johndoe@example.com",
      "name": "johndoe",
    },
  },
]
`;

exports[`User Actions Async Actions loginUser on failure 1`] = `
Array [
  Object {
    "type": "LOGIN_USER_REQUEST",
  },
  Object {
    "error": Object {
      "header": "Something wrong!",
      "kind": "wrongCredentials",
      "message": "Incorrect email (username) or password.",
    },
    "type": "LOGIN_USER_FAILURE",
  },
]
`;

exports[`User Actions Async Actions loginUser on success 1`] = `"/"`;

exports[`User Actions Async Actions loginUser on success 2`] = `
Array [
  Object {
    "type": "LOGIN_USER_REQUEST",
  },
  Object {
    "type": "LOGIN_USER_SUCCESS",
    "user": Object {
      "email": "johndoe@example.com",
      "password": "123abc",
      "username": "johndoe",
    },
  },
]
`;

exports[`User Actions Async Actions sendLogoutUser on success 1`] = `
Array [
  Object {
    "type": "LOGOUT_USER",
  },
]
`;

exports[`User Actions Async Actions setUserProfile on failure 1`] = `
Array [
  Object {
    "type": "SET_USER_PROFILE_REQUEST",
  },
  Object {
    "error": Object {
      "header": "Unknown Server Error!",
      "kind": "serverError",
      "message": "Our server may be down. Please retry later!",
    },
    "type": "SET_USER_PROFILE_FAILURE",
  },
]
`;

exports[`User Actions Async Actions setUserProfile on success 1`] = `
Array [
  Object {
    "type": "SET_USER_PROFILE_REQUEST",
  },
  Object {
    "profile": Object {
      "avatar": "https://example.com/profile_pic",
      "bio": "Web Developer",
      "initials": "JD",
      "name": "John Doe",
      "username": "johndoe",
    },
    "type": "SET_USER_PROFILE_SUCCESS",
  },
]
`;

exports[`User Actions Async Actions signupUser failure -> unknown server error 1`] = `
Array [
  Object {
    "type": "SIGNUP_USER_REQUEST",
  },
  Object {
    "error": Object {
      "header": "Unknown Server Error!",
      "kind": "serverError",
      "message": "Our server may be down. Please retry later!",
    },
    "type": "SIGNUP_USER_FAILURE",
  },
]
`;

exports[`User Actions Async Actions signupUser failure -> user already exists 1`] = `
Array [
  Object {
    "type": "SIGNUP_USER_REQUEST",
  },
  Object {
    "error": Object {
      "header": "User already exists!",
      "kind": "userExists",
      "message": "Email is already in use by another account.",
    },
    "type": "SIGNUP_USER_FAILURE",
  },
]
`;

exports[`User Actions Async Actions signupUser on success 1`] = `"/"`;

exports[`User Actions Async Actions signupUser on success 2`] = `
Array [
  Object {
    "type": "SIGNUP_USER_REQUEST",
  },
  Object {
    "type": "SIGNUP_USER_SUCCESS",
    "user": Object {
      "email": "johndoe@example.com",
      "name": "John Doe",
      "password": "123abc",
    },
  },
]
`;
